src/citrine/resources/dataset.py

```diff
@@ -265,7 +265,11 @@ def delete(self, uid: Union[UUID, str, LinkByUID, DataConcepts], *, dry_run=Fals
             Dry run is intended to be used for validation. Default: false
 
         """
-        return self.gemd.delete(uid, dry_run=dry_run)
+        if isinstance(uid, DataConcepts):
+            collection = self.gemd._collection_for(uid)
+        else:
+            collection = self.gemd
+        return collection.delete(uid, dry_run=dry_run)
```

Comment:
dataset.delete will route through the object route if we have a type, and through `storables` if not.

Reply:
why?